### PR TITLE
[IMP] Add CI to cugraph_dgl

### PR DESCRIPTION
**Pull request by VibhuJawa**
This PR replaces https://github.com/rapidsai/cugraph/pull/3293 .

### CODE REVIEW

**Review by alexbarghi-nv — APPROVED** [review]
👍

File: ci/test_python.sh [hunk 1]
```diff
@@ -86,6 +86,53 @@ pytest \
   cugraph/pytest-based/bench_algos.py
 popd
 
+if [[ "${RAPIDS_CUDA_VERSION}" == "11.8.0" ]]; then
```
Comment (alexbarghi-nv) [inline ci/test_python.sh:89]: should not run this on ARM64

File: ci/test_python.sh [hunk 2]
```diff
@@ -86,6 +86,53 @@ pytest \
...
+  rapids-mamba-retry install \
+    --force-reinstall \
+    --channel "${PYTHON_CHANNEL}" \
+    --channel pytorch \
+    --channel pytorch-nightly \
+    --channel dglteam/label/cu117 \
+    --channel nvidia \
+    cugraph-dgl \
+    dgl \
+    'pytorch::pytorch>=2.0' \
```
Comment (alexbarghi-nv) [inline ci/test_python.sh:109]: should be changed to 'pytorch>=2.0'

File: ci/test_python.sh (hunk 1 above)
Comment (VibhuJawa) [reply ci/test_python.sh:89]: Done

File: ci/test_python.sh (hunk 2 above)
Comment (VibhuJawa) [reply ci/test_python.sh]: Done

**Comment by VibhuJawa** [comment]
@rlratzel , @alexbarghi-nv . This PR is ready for review

**Comment by VibhuJawa** [comment]
@rlratzel and @ajschmidt8 could you please review this PR?

**Review by rlratzel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**